### PR TITLE
Ftr: Develop req timeout config

### PR DESCRIPTION
**Pull request by pantianying**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by zouyx** [comment]
does this pr add request timeout for each remote method?

File: config/reference_config.go [hunk 1]
```diff
@@ -168,6 +169,10 @@ func (refconfig *ReferenceConfig) getUrlMap() url.Values {
 	urlMap.Set(constant.VERSION_KEY, refconfig.Version)
 	urlMap.Set(constant.GENERIC_KEY, strconv.FormatBool(refconfig.Generic))
 	urlMap.Set(constant.ROLE_KEY, strconv.Itoa(common.CONSUMER))
+	if refconfig.RequestTimeout != "" {
+		urlMap.Set(constant.TIMEOUT_KEY, refconfig.RequestTimeout)
```
Comment (hxmhlt) [inline config/reference_config.go]: len(refconfig.RequestTimeout)==0

File: protocol/dubbo/dubbo_protocol.go [hunk 2]
```diff
@@ -67,9 +69,17 @@ func (dp *DubboProtocol) Export(invoker protocol.Invoker) protocol.Exporter {
 }
 
 func (dp *DubboProtocol) Refer(url common.URL) protocol.Invoker {
+	//default
+	requestTimeout := config.GetConsumerConfig().RequestTimeout
+
```
Comment (hxmhlt) [inline protocol/dubbo/dubbo_protocol.go]: var requestTimeout ;

**Comment by pantianying** [comment]
> does this pr add request timeout for each remote method?

yes

**Comment by flycash** [comment]
LGTM

**Comment by flycash** [comment]
LGTM

File: protocol/dubbo/dubbo_protocol.go [hunk 3]
```diff
@@ -67,9 +69,17 @@ func (dp *DubboProtocol) Export(invoker protocol.Invoker) protocol.Exporter {
 }
 
 func (dp *DubboProtocol) Refer(url common.URL) protocol.Invoker {
+	//default requestTimeout
+	var requestTimeout = config.GetConsumerConfig().RequestTimeout
+
+	requestTimeoutStr := url.GetParam(constant.REQUEST_TIMEOUT, config.GetConsumerConfig().Request_Timeout)
```
Comment (fangyincheng) [inline protocol/dubbo/dubbo_protocol.go]: when is REQUEST_TIMEOUT  set to `url`

Comment (pantianying) [reply protocol/dubbo/dubbo_protocol.go]: image

**Pull request merged**